src/slices/tasksSlice.js

```diff
@@ -0,0 +1,76 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+import { saveTasks, loadTasks } from '../repositories/taskRepository';
+
+const initialState = {
+  tasks: loadTasks().tasks || [],
+  tasksOrder: loadTasks().tasksOrder || [],
+};
+
+const tasksSlice = createSlice({
+  name: 'tasks',
+  initialState,
+  reducers: {
+    addTask(state, action) {
+      const newId = uuidv4();
+      state.tasks.push({ ...action.payload, id: newId });
+      state.tasksOrder.push(newId);
+
+      saveTasks(state);
```

Comment:
Working with localStorage have side effects. It is not good to have side effects in reducers. Either move the work with localStorage to Redux middleware or create an action with https://redux-toolkit.js.org/api/createAsyncThunk

Reply:
created localStorageMiddleware